src/xdp/xsk.c

```diff
@@ -2947,52 +3082,50 @@ XskSockoptSetUmem(
         goto Exit;
     }
 
-    //
-    // If support is needed for kernel mode AF_XDP sockets, UMEM MDL setup
-    // needs more thought.
-    //
-    ASSERT(RequestorMode == UserMode);
-
-    Umem->Mapping.Mdl =
-        IoAllocateMdl(
-            Umem->Reg.Address,
-            (ULONG)Umem->Reg.TotalSize,
-            FALSE, // SecondaryBuffer
-            FALSE, // ChargeQuota
-            NULL); // Irp
-    if (Umem->Mapping.Mdl == NULL) {
-        Status = STATUS_INSUFFICIENT_RESOURCES;
-        goto Exit;
-    }
+    if (RequestorMode != KernelMode) {
```

Comment:
Looks like we check if requestor is kernel more than once. Should we be worried about that, and try to optimize it, or not?

Reply:
This is control path so I'm not worried about this extra check